### ISSUE TITLE
WorkerPro `ttl` doesn't work with sandboxed jobs

### ISSUE THREAD

**Issue by rosslavery** [taskforcesh/bullmq-pro-support#20]
Returning an `Observable` from inside a sandboxed worker prevents the following check from working as expected:

```ts
// worker-pro.js#callProcessJob
const result = await this.processFn(job, token);
if (isObservable(result)) {
...
```

Instead, the `typeof` is simply `object` with it's value being `{}`.

This prevents the subsequent code that handles TTL from cancelling the job as expected.

Validated that un-sandboxing the job makes it work as expected.

**Comment by manast** [comment]
Yes, unfortunatelly Sandboxed processors are not yet up to the same features as normal processors, we should document this better.

**Comment by jingom368** [comment]
Could you please let me know how to apply TTL in WorkerPro by returning an Observable from the sandbox Processor? Additionally, has this issue been resolved?

**Comment by manast** [comment]
As it is not trivial to add observables to sandboxed processors, but on the other hand a sandboxed processor can actually kill the process or thread running the job, I think we can deliver a real job cancelation without using observables, I think this would help most users, still before cancellation the job could get a signal to notify it is going to be cancelled in a given number of seconds so that it has a chance to do a graceful exit.
